### ISSUE TITLE
suggest new name to repo

### ISSUE THREAD

**Issue by qballer** [teambit/envs#38]
Environments doesn't mean a whole lot to people when we discuss them. Envs  - short for that, even less. I wonder what will be the best name for these tools created in this repo. These tools are made around the idea of running an activity in a capsule. A name which popped to mind was "shells". Different command line tools which you can run over capsule makes some sense to me. It also a wrapper over the capsule - a shell. 

Downside to this is that term is overloaded something which is out there. Other offers would be awesome.

**Comment by imsnif** [comment]
Plugins, extensions, addons.

**Comment by Tallyb** [comment]
gizmos

**Comment by qballer** [comment]
@imsnif  Here we march in to our very old discussion of what are environments? If I write an extension which extends  bit add or export it would not be an env in my mind. Envs run in the context of isolated components. Not all things are like that. I want the name to give some insight to what these tools do - builders, testers, linters, coverage tools etc. 

@Tallyb gizmos made me smile. Still lacking in terms of intuition to what these packages do - no?

**Comment by Tallyb** [comment]
@qballer - you smile and the world smiles with you... gizmo does whatever you tell him to do.

**Comment by davidfirst** [comment]
@qballer , I see where you're coming from, in my mind, I have this distinction as well. However, I'm unsure about it.

> If I write an extension which extends bit add or export it would not be an env in my mind

If you think about it, all these tools, builders, tester, linters extend "bit tag", "bit ci-update" etc. Builders and tester also have their own commands "bit build" and "bit test" but these commands probably should have been part of the extension and not part of Bit. 
You could write a linter as an extension and ship it with a new command "bit lint". 
So, basically, extensions hook into Bit ecosystem, extend Bit, can have their own commands, can extend other commands. 
From Bit perspective, It is hard to make the distinction between whether you extend Bit in order to do code-analysis or you extend Bit to add files in a different way. 
If you look at VSC as an example, some extensions are entire language support (php, typescript, etc.) and some do minor things, still, all are extensions without the distinction above.

**Comment by qballer** [comment]
I understand that there is no difference in the meaning of bit. All are just extensions. But with these extension group the user can do most of the heavy dev lifting  - testing, building, formatting etc.

**Comment by davidfirst** [comment]
I guess we should ask ourselves why the distinction between the tools above and other extensions are needed. 
If it's hard to come up with a definition of these tools, maybe it's better not to make the distinction.

In this post I see the following descriptions:

> These tools are made around the idea of running an activity in a capsule

Linters and static-code-analysis tools probably don't need capsule and these tools you do want to include. Formatters (e.g. prettier) for sure don't need capsule.

> with these extension group the user can do most of the heavy dev lifting - testing, building, formatting etc

It's hard to put the line what's considered a heavy dev lifting. In any case, they probably won't map 1:1 with the ones that need capsule.

So, I believe, if we could come up with a better definition, it'd help. If we couldn't, probably we shouldn't.

**Comment by qballer** [comment]
Ok. So which concrete name? what @imsnif suggested?
